models.ts

```diff
@@ -6,13 +6,68 @@ class Peli {
   title: string;
   tags: string[];
 }
-
 class PelisCollection {
-  getAll(): Promise<Peli[]> {
-    return jsonfile("...laRutaDelArchivo").then(() => {
-      // la respuesta de la promesa
-      return [];
-    });
+  data: Peli[] = [];
+  
+  getAll():Promise<Peli[]> {
+    const promesa = jsonfile.readFile("./pelis.json");
+    promesa.then((res=>{
+      return this.data = res;
+
+    }))
+    return promesa;
   }
+
+  getById(id:number): Promise <any>{
+    const promesaNueva =this.getAll().then(() =>{
+       return this.data.find(item => item.id == id)
+       
+    })
+    return promesaNueva;
+    
+        
+      
+    };
+
+    search(options: any):Promise <any>{
+      let promesa = this.getAll().then(() =>{
+        let listaDePeliculas = this.data;
+        for(const k in options){
```

Comment:
Acá quizás ahorraría todo el trabajo del for, preguntando directamente con un if(options.title / options.tag).